raft.c

```diff
@@ -2391,28 +2494,14 @@ void handleShardGroupUpdate(RedisRaftCtx *rr, RaftReq *req)
         goto exit;
     }
 
-    /* TODO: validate/handle slot range changes:
-     *  i.e. preventing stable overlap switching to migrating or adding as importing
-     */
-    ShardGroup *sg = getShardGroupById(rr, req->r.shardgroup_add.id);
-    if (sg == NULL) {
+    if (ShardGroupsAppendLogEntry(rr, req->r.shardgroups_replace.len, req->r.shardgroups_replace.shardgroups,
```

Comment:
In line 2493, `RedisModule_ReplyWithError(req->ctx, "failed, please check logs.");` is unnecessary as we already set response in  `checkLeader()` or `checkRaftState()`. Maybe we use this oppurtunity and remove that line as part of this PR.

Reply:
not 100% sure I understand the comment.  I see we dont need the reply with error in the `checkLeader()` / `CheckRaftState()` conditional failure, but I still think its needed here.

unless that's what you mean.

Reply:
Yeah, I meant that. I see you already removed that line.